windows/src/engine/inst-cpp/insthelper.cpp

```diff
@@ -0,0 +1,211 @@
+#include "insthelper.h"
+#include "pch.h"
+#include <AccCtrl.h>
+#include <ShlObj_core.h>
+#include <iostream>
+#include <knownfolders.h>
+#include <limits.h>
+#include <msiquery.h>
+#include <security.h>
+#include <utility>
+#include <tchar.h>
+#include <string>
+#include <AclAPI.h>
+#include <Windows.h>
+#include <Msi.h>
+#include <initguid.h>
+#include <comdef.h>
+#include <msctf.h>
+#include <atlbase.h>
+#include<sstream>
+#include<iomanip>
+
+const LPCTSTR SFolderKeymanRoot = TEXT("\\Keyman");
+
+extern "C" __declspec(dllexport) unsigned int EnginePostInstall(MSIHANDLE hInstall) {
+  HANDLE hFile;
+
+  // Find %appdata% path
+  PWSTR path = new wchar_t[MAX_PATH];
```

Comment:
You might as well put path on the stack so you don't have to worry about freeing it:

```suggestion
  // Find %appdata% path
  wchar_t path[MAX_PATH];
```

Reply:
Right!